adafruit_circuitplayground/circuit_playground_base.py

```diff
@@ -689,23 +690,38 @@ def red_led(self, value):
     @staticmethod
     def _sine_sample(length):
         tone_volume = (2**15) - 1
+        # Amplitude shift up in order to not have negative numbers
         shift = 2**15
         for i in range(length):
             yield int(tone_volume * math.sin(2 * math.pi * (i / length)) + shift)
 
-    def _generate_sample(self, length=100):
+    @staticmethod
+    def _square_sample(length):
+        # Square waves are MUCH louder than then sine
+        tone_volume = (2**16) - 1
+        half_length = length // 2
+        for _ in range(half_length):
+            yield tone_volume
+        for _ in range(half_length):
+            yield 0
+
+    def _generate_sample(self, length=100, waveform="sine"):
```

Comment:
waveform could take constant values instead of a string. 

I don't know if strictly better one way or the other but I tend to try to use constant number values with descriptively named variables instead of strings when I write code like this. 

I'm not sure my tendencies are worth holding this up over though so we can see if anyone else has thoughts on it.

i.e usage would look something like:
```py
circuitplayground.start_tone(234, circuitplayground.SINE_WAVEFORM)
```
and the value can then be a constant int rather than a string.

Reply:
That's a great idea as well.  Is there an enum type in circuit python?

Reply:
I don't think we have enum, but we do have a const that is inherited from micropython: https://docs.circuitpython.org/en/latest/docs/library/micropython.html?#micropython.const

Reply:
since the variable names may take some space as well it may be worthwhile to go with a more shorthand version like `SINE_WAVE` instead of `SINE_WAVEFORM` as well. Every bit counts to an extent here.

Reply:
That is a good idea.  I will look over variable names.

Reply:
`SINE_WAVEFORM`, etc. are going to be the equivalent of strings, so you will not gain space, unfortunately, I think.